Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -1381,26 +1381,106 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
+                    if (!accessibilityNode.shouldBeDisplayed() || this._elementCloseTag) {
+                        this.hasChildren = false;
+                        if (this.parent) {
+                            this.parent.removeChild(this);
+                        }
+                        return;
+                    }
+                    
+                    let baseURL = domNode.frame ? domNode.frame.url : null;
+                    let titleDOM = this._buildAccessibilityDOM(accessibilityNode, baseURL);
+                    this.title.textContent = ""; 
+                    this.title.appendChild(titleDOM);
+                });
+
+            }
+            else{
+                this.title.appendChild(this._nodeTitleInfo().titleDOM);
+                this._createBadges();
+            }
             this._highlightResult = undefined;
         }
-        this._createBadges();
-
-        // Setting this.title will implicitly remove all children. Clear the
-        // selection element so that we properly recreate it if necessary.
         this._selectionElement = null;
         this.updateSelectionArea();
         this._highlightSearchResults();
         this._updatePseudoClassIndicator();
         this._updateBreakpointStatus();
     }
 
+    _buildAccessibilityDOM(accessibilityNode, baseURL)
+    {
+        var domNode = accessibilityNode.domNode;
+        var titleDOM = document.createDocumentFragment();
+        var tagElement = document.createElement("span");
+        tagElement.className = "accessibility-tree-node";
+        if (accessibilityNode.isIgnored) 
+            tagElement.classList.add("disabled-element");
+        
+        var tagNameElement = document.createElement("span");
+        let roleSpan = document.createElement("span");
+        roleSpan.className = "accessibility-role";
+        roleSpan.textContent = accessibilityNode.computedRole.charAt(0).toUpperCase() + accessibilityNode.computedRole.slice(1);
+        tagNameElement.appendChild(roleSpan);
+        if (accessibilityNode.computedLabel) {
+            let colon = document.createTextNode(": ");
+            let labelSpan = document.createElement("span");
+            labelSpan.className = "accessibility-label";
+            labelSpan.textContent = `"${accessibilityNode.computedLabel}"`;
+            tagNameElement.appendChild(colon);
+            tagNameElement.appendChild(labelSpan);
+        }
+        if (typeof accessibilityNode.focused === "boolean") {
+            let focusSpan = document.createElement("span");
+            focusSpan.className = "accessibility-focus";
+            focusSpan.textContent = WI.UIString(" Focused: %s").format(accessibilityNode.focused);
+            tagNameElement.appendChild(focusSpan);
+        }
+        var attributes = domNode.attributes();
+        for (var i = 0; i < attributes.length; ++i) {
+            var attr = attributes[i];
+            if (attr.name === "src" || /\bhref\b/.test(attr.name)) {
```

Comment:
This seems a little worrying, but it's probably fine if it's needed for accessibility tree only. Is it what assistive tech actually sees, though?